### PR TITLE
Update SQL objects in accordance with new downsampling UX

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Note for reviewers: For description regarding the new UX, please read the description at https://github.com/timescale/promscale/pull/1808

## Merge requirements

Please take into account the following non-code changes that you may need to make with your PR:

- [ ] CHANGELOG entry for user-facing changes
- [ ] Updated the relevant documentation

### CODE REVIEW

**Review by jgpruitt — CHANGES_REQUESTED** [review]

File: sql-tests/testdata/metric_downsampling.sql [hunk 1]
```diff
@@ -0,0 +1,47 @@
+\set ON_ERROR_STOP 1
```
[the remark this answers is not in the document]
Comment (Harkishen-Singh) [reply sql-tests/testdata/metric_downsampling.sql:1]: I am not sure if pgtap is helpful for this test. The test mostly confirms the table structure of `dowmsample`, `metric_downsample` and related metrics, so it was easier to use snapshots.

**Comment by Harkishen-Singh** [comment]
Note: I have added another commit `Add querying view & support for dowmsampling metrics with no metadata` that is not primarily related to this PR. I wanted to avoid conflicts between creating another PR (from feature_metric_rollup branch) and this PR getting merged, hence reused this PR itself.

File: migration/idempotent/017-metric-downsampling.sql [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+$$
+BEGIN
+    EXECUTE FORMAT($sql$
+        ALTER MATERIALIZED VIEW %1$I.%2$I SET (timescaledb.compress = true)
+    $sql$, _schema, _table_name);
+END;
+$$
+LANGUAGE PLPGSQL;
+
+-- This creates a view that acts as a interface between connector and the downsampled data while querying.
```
Comment (Harkishen-Singh) [inline migration/idempotent/017-metric-downsampling.sql:83]: This is the new part added in second commit.

File: migration/idempotent/017-metric-downsampling.sql [hunk 3]
```diff
@@ -0,0 +1,392 @@
...
+AS
+$$
+DECLARE
+    _metric_type TEXT;
+    _default_query_column TEXT := 'sum / count';
+
+BEGIN
+   SELECT type INTO _metric_type FROM _prom_catalog.metadata WHERE metric_family = _metric_name;
+    IF (_metric_type IS NULL) THEN
+        -- Guess the metric type based on the metric/table suffix.
```
Comment (Harkishen-Singh) [inline migration/idempotent/017-metric-downsampling.sql:119]: This too as part of second commit.

File: migration/idempotent/017-metric-downsampling.sql [hunk 4]
```diff
@@ -0,0 +1,392 @@
...
+-- maps to the right column of downsampled data.
+CREATE OR REPLACE FUNCTION _prom_catalog.create_default_downsampling_query_view(_schema TEXT, _table_name TEXT, _default_column TEXT)
+RETURNS VOID
+    SET search_path = pg_catalog, pg_temp
+AS
+$$
+DECLARE
+    -- We prefix with q_ to avoid collision with names of materialized views. This prefix can be added
+    -- on the connector side.
+    query TEXT := 'CREATE OR REPLACE VIEW %1$I.q_%2$I AS SELECT time, series_id, %3$s AS value FROM %1$I.%2$I';
```
Comment (Harkishen-Singh) [inline migration/idempotent/017-metric-downsampling.sql:96]: See commit [Adds support to query default downsampling views in absence of `__column__`](https://github.com/timescale/promscale/pull/1808/commits/7464174d04281b2e81652fa8c1c6c08b13378075) for understanding the use of this view.

**Comment by niksajakovljevic** [comment]
It would be great to rename `resolution` to `interval` so it's in line with connector naming.

**Pull request merged**